applications/ALSAFullDuplex.C

```diff
@@ -133,7 +139,13 @@ int main(int argc, char *argv[]) {
 	if (res<0)
 		return res;
 
+  string formatStr;
 	snd_pcm_format_t format=SND_PCM_FORMAT_S32_LE;
+
+  if (op.getArg<string>("f", argc, argv, formatStr, i=0)!=0){
+     format = snd_pcm_format_value(formatStr.c_str());
+  }
+
 	if ((res=fullDuplex.setFormat(format))<0)
```

Comment:
Use the gtkiostream ALSA::Hardware::setFormat method which takes in the string.
https://github.com/flatmax/gtkiostream/blob/master/include/ALSA/Hardware.H#L214